source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -118,10 +119,20 @@ namespace RendererQml
         rectangle->Property("Layout.preferredHeight", "bodyLayout.height");
         rectangle->AddChild(bodyLayout);
 
-        AddContainerElements(bodyLayout, card->GetBody(), context);
-        AddActions(bodyLayout, card->GetActions(), context);
+		if (!card->GetBody().empty())
+		{
+			CheckLastBodyElementIsShowCard(card->GetBody().back(), context);
+		}
```

Comment:
You should do the if check inside `CheckLastBodyElementIsShowCard`. Would be cleaner here. Just pass `card->GetBody()` and let eh function do the validations.

Reply:
Thanks for pointing out. I'll make this change